src/server/saved_objects/client/lib/create_id_query.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * Finds a document by either its v5 or v6 format
+ *
+ * @param type The documents type
+ * @param id The documents id or legacy id
+**/
+export function createIdQuery({ type, id }) {
```

Comment:
This could potentially return an unmigrated v5 type that had the id of 'dashboard:something' when the user was looking for an item with id of 'something'